RunETL/runETL.sh

```diff
@@ -1,48 +1,72 @@
-diagnosis_filename=${diagnosis_filename:-diagnosis_codes.csv}
-read -p "Drug lookup filename [drug_codes.csv]: " drug_filename
-drug_filename=${drug_filename:-drug_codes.csv}
-read -p "Tissue lookup filename [tissue_codes.csv]: " tissue_filename
-tissue_filename=${tissue_filename:-tissue_codes.csv}
-read -p "Other therapies lookup filename [muc_therapy_other_therapies.csv]: " other_therapies_filename
-other_therapies_filename=${other_therapies_filename:-muc_therapy_other_therapies.csv}
-read -p "Name of your site or data set [ESFURN]: " esfurn_site
-esfurn_site=${esfurn_site:-ESFURN}
-
-sed -i -e "s@data_folder@$data_folder@g" docker-compose.yml
-sed -i -e "s/db_username/$db_username/g" docker-compose.yml
-sed -i -e "s/db_password/$db_password/g" docker-compose.yml
-sed -i -e "s/verbosity_level/$verbosity_level/g" docker-compose.yml
-sed -i -e "s/image_tag/$image_tag/g" docker-compose.yml
-sed -i -e "s/date_last_export/$date_last_export/g" docker-compose.yml
-sed -i -e "s/input_filename/$input_filename/g" docker-compose.yml
-sed -i -e "s/diagnosis_filename/$diagnosis_filename/g" docker-compose.yml
-sed -i -e "s/drug_filename/$drug_filename/g" docker-compose.yml
-sed -i -e "s/tissue_filename/$tissue_filename/g" docker-compose.yml
-sed -i -e "s/other_therapies_filename/$other_therapies_filename/g" docker-compose.yml
-sed -i -e "s/esfurn_site/$esfurn_site/g" docker-compose.yml
-
-docker login harbor.esfurn.org
-docker compose pull
-docker compose run --rm --name etl etl
+#!/usr/bin/env bash
+set -e
+
+#############
+## RUN ETL ##
+#############
+REGISTRY_ETL_RUNNER=harbor.honeur.org
+REPOSITORY_ETL_RUNNER=library
+IMAGE_ETL_RUNNER=etl-runner
+VERSION_ETL_RUNNER=1.1.4
+TAG_ETL_RUNNER=$VERSION_ETL_RUNNER
+
+LOG_FOLDER_HOST=${PWD}/log
+DATA_FOLDER_HOST=${PWD}/data
+QA_FOLDER_HOST=${PWD}/qa
+
+echo "Pull ETL runner image"
+docker pull $REGISTRY_ETL_RUNNER/$REPOSITORY_ETL_RUNNER/$IMAGE_ETL_RUNNER:$TAG_ETL_RUNNER
+
+REGISTRY=harbor.esfurn.org
+REPOSITORY=etl
+CDM_SCHEMA=omopcdm
+VOCAB_SCHEMA=$CDM_SCHEMA
+
+echo "Download ETL questions"
+curl -fsSL https://raw.githubusercontent.com/solventrix/ESFURN-Setup/blob/master/RunETL/questions.json --output questions.json
+
+touch etl-runner.env
+# etl image
+echo "THERAPEUTIC_AREA=esfurn" >> etl-runner.env
+echo "REGISTRY=$REGISTRY" >> etl-runner.env
+echo "ETL_IMAGE_NAME=$REPOSITORY/esfurn_etl" >> etl-runner.env
+echo "ETL_IMAGE_TAG=current" >> etl-runner.env
+# logs
+echo "LOG_FOLDER_HOST=$LOG_FOLDER_HOST" >> etl-runner.env
+echo "LOG_FOLDER=/log" >> etl-runner.env
+echo "VERBOSITY_LEVEL=$VERBOSITY_LEVEL" >> etl-runner.env
+# schemas 
+echo "DB_SCHEMA=$CDM_SCHEMA" >> etl-runner.env
+echo "VOCAB_SCHEMA=$VOCAB_SCHEMA" >> etl-runner.env 
+# site 
+echo "ESFURN_SITE=$ESFURN_SITE" >> etl-runner.env 
+# source data
+echo "DATA_FOLDER_HOST=$DATA_FOLDER_HOST" >> etl-runner.env
+echo "DATA_FOLDER=/data" >> etl-runner.env
+echo "INPUT_DATA=$INPUT_DATA" >> etl-runner.env
```

Comment:
Remove the variables that are not defined in this run script but are defined using the questions.json